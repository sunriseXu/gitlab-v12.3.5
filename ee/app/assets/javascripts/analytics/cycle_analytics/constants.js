import { __ } from '~/locale';

export const DEFAULT_DATA_TIME_FRAME = 30;

export const EVENTS_LIST_ITEM_LIMIT = 50;

export const EMPTY_STAGE_TEXT = {
  issue: __(
    'The issue stage shows the time it takes from creating an issue to assigning the issue to a milestone, or add the issue to a list on your Issue Board. Begin creating issues to see data for this stage.',
  ),
  plan: __(
    'The planning stage shows the time from the previous step to pushing your first commit. This time will be added automatically once you push your first commit.',
  ),
  code: __(
    'The coding stage shows the time from the first commit to creating the merge request. The data will automatically be added here once you create your first merge request.',
  ),
  test: __(
    'The testing stage shows the time GitLab CI takes to run every pipeline for the related merge request. The data will automatically be added after your first pipeline finishes running.',
  ),
  review: __(
    'The review stage shows the time from creating the merge request to merging it. The data will automatically be added after you merge your first merge request.',
  ),
  staging: __(
    'The staging stage shows the time between merging the MR and deploying code to the production environment. The data will be automatically added once you deploy to production for the first time.',
  ),
  production: __(
    'The production stage shows the total time it takes between creating an issue and deploying the code to production. The data will be automatically added once you have completed the full idea to production cycle.',
  ),
};
