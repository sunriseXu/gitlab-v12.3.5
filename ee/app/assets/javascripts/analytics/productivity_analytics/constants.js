import { __, s__ } from '~/locale';

export const chartKeys = {
  main: 'main',
  timeBasedHistogram: 'timeBasedHistogram',
  commitBasedHistogram: 'commitBasedHistogram',
  scatterplot: 'scatterplot',
};

export const chartTypes = {
  histogram: 'histogram',
  scatterplot: 'scatterplot',
};

export const metricTypes = [
  {
    key: 'time_to_first_comment',
    label: __('Time from first commit until first comment'),
    chart: chartKeys.timeBasedHistogram,
  },
  {
    key: 'time_to_last_commit',
    label: __('Time from first comment to last commit'),
    chart: chartKeys.timeBasedHistogram,
  },
  {
    key: 'time_to_merge',
    label: __('Time from last commit to merge'),
    chart: chartKeys.timeBasedHistogram,
  },
  {
    key: 'commits_count',
    label: __('Number of commits per MR'),
    chart: chartKeys.commitBasedHistogram,
  },
  {
    key: 'loc_per_commit',
    label: __('Number of LOCs per commit'),
    chart: chartKeys.commitBasedHistogram,
  },
  {
    key: 'files_touched',
    label: __('Number of files touched'),
    chart: chartKeys.commitBasedHistogram,
  },
];

export const tableSortOrder = {
  asc: {
    title: s__('ProductivityAnalytics|Ascending'),
    value: 'asc',
    icon: 'sort-lowest',
  },
  desc: {
    title: s__('ProductivityAnalytics|Descending'),
    value: 'desc',
    icon: 'sort-highest',
  },
};

export const daysToMergeMetric = {
  key: 'days_to_merge',
  label: s__('ProductivityAnalytics|Days to merge'),
};

export const defaultMaxColumnChartItemsPerPage = 20;

export const maxColumnChartItemsPerPage = {
  [chartKeys.main]: 40,
};

export const dataZoomOptions = [
  {
    type: 'slider',
    bottom: 10,
    start: 0,
  },
  {
    type: 'inside',
    start: 0,
  },
];

/**
 * #418cd8 --> $blue-400 (see variables.scss)
 */
export const columnHighlightStyle = { color: '#418cd8', opacity: 0.8 };

export const accessLevelReporter = 20;

export const projectsPerPage = 50;
