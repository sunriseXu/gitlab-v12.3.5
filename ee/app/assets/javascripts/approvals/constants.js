export const TYPE_USER = 'user';
export const TYPE_GROUP = 'group';
export const TYPE_HIDDEN_GROUPS = 'hidden_groups';

export const RULE_TYPE_FALLBACK = 'fallback';
export const RULE_TYPE_REGULAR = 'regular';
export const RULE_TYPE_CODE_OWNER = 'code_owner';
