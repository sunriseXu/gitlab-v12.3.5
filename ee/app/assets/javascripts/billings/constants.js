export const CUSTOMER_PORTAL_URL = 'https://customers.gitlab.com/subscriptions';
export const TABLE_TYPE_DEFAULT = 'default';
export const TABLE_TYPE_FREE = 'free';
export const TABLE_TYPE_TRIAL = 'trial';
