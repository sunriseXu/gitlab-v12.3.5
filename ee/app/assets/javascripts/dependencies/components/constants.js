export const WARNING = 'warning';
export const DANGER = 'danger';

export const WARNING_ALERT_CLASS = 'warning_message';
export const DANGER_ALERT_CLASS = 'danger_message';

export const WARNING_TEXT_CLASS = 'text-warning-900';
export const DANGER_TEXT_CLASS = 'text-danger-900';

// Limit the number of vulnerabilities to display so as to avoid jank.
// In practice, this limit will probably never be reached, since the
// largest number of vulnerabilities we've seen one dependency have is 20.
export const MAX_DISPLAYED_VULNERABILITIES_PER_DEPENDENCY = 50;
