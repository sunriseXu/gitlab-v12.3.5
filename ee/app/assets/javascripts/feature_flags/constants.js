import _ from 'underscore';

export const ROLLOUT_STRATEGY_ALL_USERS = 'default';
export const ROLLOUT_STRATEGY_PERCENT_ROLLOUT = 'gradualRolloutUserId';
export const ROLLOUT_STRATEGY_USER_ID = 'userWithId';

export const PERCENT_ROLLOUT_GROUP_ID = 'default';

export const DEFAULT_PERCENT_ROLLOUT = '100';

export const ALL_ENVIRONMENTS_NAME = '*';

export const INTERNAL_ID_PREFIX = 'internal_';

export const fetchPercentageParams = _.property(['parameters', 'percentage']);
export const fetchUserIdParams = _.property(['parameters', 'userIds']);
