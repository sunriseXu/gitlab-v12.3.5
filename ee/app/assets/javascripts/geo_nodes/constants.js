export const NODE_ACTIONS = {
  TOGGLE: 'toggle',
  REMOVE: 'remove',
};

export const VALUE_TYPE = {
  PLAIN: 'plain',
  GRAPH: 'graph',
  CUSTOM: 'custom',
};

export const CUSTOM_TYPE = {
  SYNC: 'sync',
  EVENT: 'event',
  STATUS: 'status',
};

export const HEALTH_STATUS_ICON = {
  healthy: 'status_success',
  unhealthy: 'status_failed',
  disabled: 'status_canceled',
  unknown: 'status_warning',
  offline: 'status_canceled',
};

export const TIME_DIFF = {
  FIVE_MINS: 300,
  HOUR: 3600,
};

export const STATUS_DELAY_THRESHOLD_MS = 60000;

export const HELP_INFO_URL =
  'https://docs.gitlab.com/ee/administration/geo/disaster_recovery/background_verification.html#repository-verification';
