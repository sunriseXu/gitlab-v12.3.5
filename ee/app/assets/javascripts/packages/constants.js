const PackageType = {
  MAVEN: 'maven',
  NPM: 'npm',
};

export default PackageType;
