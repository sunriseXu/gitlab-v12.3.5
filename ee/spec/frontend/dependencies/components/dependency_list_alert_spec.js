import { createLocalVue, shallowMount } from '@vue/test-utils';
import { WARNING } from 'ee/dependencies/components/constants';
import DependencyListAlert from 'ee/dependencies/components/dependency_list_alert.vue';

describe('DependencyListAlert component', () => {
  let wrapper;

  const factory = (props = {}) => {
    const localVue = createLocalVue();

    wrapper = shallowMount(localVue.extend(DependencyListAlert), {
      localVue,
      sync: false,
      propsData: { ...props },
      slots: {
        default: '<p>foo <span>bar</span></p>',
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe('given no props', () => {
    beforeEach(() => {
      factory();
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('given the warning type and headerText props', () => {
    beforeEach(() => {
      factory({ type: WARNING, headerText: 'Some header' });
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('given the headerText prop', () => {
    beforeEach(() => {
      factory({ headerText: 'A header' });
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('clicking on the close button', () => {
    beforeEach(() => {
      factory();
      wrapper.find('.js-close').vm.$emit('click');
      return wrapper.vm.$nextTick();
    });

    it('emits the close event', () => {
      expect(wrapper.emitted().close.length).toBe(1);
    });
  });
});
