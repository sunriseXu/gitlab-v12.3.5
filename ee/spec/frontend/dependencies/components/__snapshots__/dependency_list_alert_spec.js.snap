// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DependencyListAlert component given no props matches the snapshot 1`] = `
<div
  class="danger_message text-danger-900"
>
  <glbutton-stub
    aria-label="Close"
    class="btn-blank float-right mr-1 mt-1 js-close text-danger-900"
  >
    <icon-stub
      aria-hidden="true"
      cssclasses=""
      name="close"
      size="16"
    />
  </glbutton-stub>
   
  <!---->
   
  <p>
    foo 
    <span>
      bar
    </span>
  </p>
</div>
`;

exports[`DependencyListAlert component given the headerText prop matches the snapshot 1`] = `
<div
  class="danger_message text-danger-900"
>
  <glbutton-stub
    aria-label="Close"
    class="btn-blank float-right mr-1 mt-1 js-close text-danger-900"
  >
    <icon-stub
      aria-hidden="true"
      cssclasses=""
      name="close"
      size="16"
    />
  </glbutton-stub>
   
  <h4
    class="text-danger-900"
  >
    A header
  </h4>
   
  <p>
    foo 
    <span>
      bar
    </span>
  </p>
</div>
`;

exports[`DependencyListAlert component given the warning type and headerText props matches the snapshot 1`] = `
<div
  class="warning_message text-warning-900"
>
  <glbutton-stub
    aria-label="Close"
    class="btn-blank float-right mr-1 mt-1 js-close text-warning-900"
  >
    <icon-stub
      aria-hidden="true"
      cssclasses=""
      name="close"
      size="16"
    />
  </glbutton-stub>
   
  <h4
    class="text-warning-900"
  >
    Some header
  </h4>
   
  <p>
    foo 
    <span>
      bar
    </span>
  </p>
</div>
`;
